subworkflows/ebi-metagenomics/reads_qc/main.nf

```diff
@@ -1,35 +1,108 @@
 
-include { FASTP      } from '../../../modules/ebi-metagenomics/fastp/main'
-include { SEQTK_SEQ     } from '../../../modules/ebi-metagenomics/seqtk/seq/main'
+include { SEQFU_CHECK            } from '../../../modules/ebi-metagenomics/seqfu/check/main'
+include { ASSESSMCPPROPORTIONS   } from '../../../modules/ebi-metagenomics/assessmcpproportions/main'
+include { FASTQSUFFIXHEADERCHECK } from '../../../modules/ebi-metagenomics/fastqsuffixheadercheck/main'
+include { FASTP                  } from '../../../modules/ebi-metagenomics/fastp/main'
+include { SEQTK_SEQ              } from '../../../modules/ebi-metagenomics/seqtk/seq/main'
 
 workflow  READS_QC {
 
     take:
+    filter_amplicon // channel: val(boolean)
     ch_reads    // channel: [ val(meta), [ fastq ] ]
     save_merged // channel:  val(boolean)
 
     main:
-
     ch_versions = Channel.empty()
 
-    FASTP ( ch_reads, params.save_trimmed_fail, save_merged )
+    // ***** Necessary mapping functions *****
+    filterBySeqFuStatus = { meta, seqfu_res ->
+        seqfu_check_status = seqfu_res[0]
+        if (seqfu_check_status == "OK"){
+            [ meta ]
+        }
+    }
+
+    filterBySuffixHeaderStatus = { meta, sufhd_res ->
+        if (sufhd_res.countLines() == 0){
+            [ meta ]
+        }
+    }
+
+    add_mcp_flags = { meta, fastq ->
+        [ meta, "auto", "auto", fastq ]
+    }
+
+    prepareForMCPCheck = { meta, fwd_flag, rev_flag, fastq ->
+        if (meta.single_end){
+            [ meta, fwd_flag, rev_flag, fastq ]
+        } else {
+                [ meta, fwd_flag, rev_flag, fastq[0] ]
+        }
+    }
+
+    filterByAmpliconCheck = { meta, strategy ->
+        if (strategy == "AMPLICON"){
+            [ meta ]
+        }
+    }
+    // ***** Necessary mapping functions *****
+
+    SEQFU_CHECK(ch_reads)
+    ch_versions = ch_versions.mix(SEQFU_CHECK.out.versions.first())
+
+    passed_seqfu_reads = SEQFU_CHECK.out.tsv
+        .splitCsv(sep: "\t", elem: 1)
+        .map(filterBySeqFuStatus)
```

Comment:
I don't think map can be used to filter things.. for that [`.filter` is the right operator](https://www.nextflow.io/docs/latest/reference/operator.html#filter)

I suspect this map is leaving empty values that then are filtered in the .joins, but it's a bit dirty IMO.

Reply:
Like -> https://github.com/EBI-Metagenomics/nf-modules/pull/68

Reply:
Done!